File: 2231-find-first-palindromic-string-in-the-array/find-first-palindromic-string-in-the-array.cpp
class Solution {
public:
    string firstPalindrome(vector<string>& words) {
       for(int i = 0; i <words.size();i++){
        int l = 0;
        int h = words[i].size() - 1;
        while(l <= h){
            if(words[i][l] != words[i][h]) break;
                l++; 
                h-- ;
        }
        if(l >= h) return words[i];
       }
       return ""; 
    }
};

File: 2231-find-first-palindromic-string-in-the-array/find-first-palindromic-string-in-the-array.java
class Solution {
    public String firstPalindrome(String[] words) {
        for(int i = 0; i < words.length; i++){
            int low = 0;
            int high = words[i].length() - 1;

            while(low <= high){
                if(words[i].charAt(low) != words[i].charAt(high) ) break;
                low++;
                high --;
            }
            if(low >= high) return words[i];
        }
        return "";
    }
}